Clarify nightly style policy (#180)
rust-lang/style-team: nightly-style-policy.md

> Initial PR(s) implementing new syntax filed against rust-lang/rust should

Author: s/policy/procedure in the title

> Initial PR(s) implementing new syntax filed against rust-lang/rust should

Author: make a nice opener that restates that we don't[^1] prescribe nightly style policy

[^1]: until we do, at which time we should make rustfmt follow it

Author: jane says something like "if nightly formatting is stated, then we try to follow it"?

generally leave the rustfmt subtree untouched. In cases that that need to
modify rustfmt (for example, to fix compiler errors that come from adding
new variants to AST representation), they should modify rustfmt in such a

> way to keep existing formatting verbatim.

Reviewer: Just for awareness, I will update this later with a direct link to a "how to" rustfmt doc on this once we have one, refs https://github.com/rust-lang/rustfmt/issues/5770

Rustfmt is allowed to implement nightly-only formatting behavior without that

> syntax being specified by the style guide. Initial PR implementors are 
> encouraged but not required to open a PR against rustfmt suggesting an initial

Reviewer: A few small changes suggested all as one. Don't feel strongly about any, feel free to take/leave any/all
```suggestion
syntax being specified by the style guide. The initial authors of PR implementing new features in rust-lang/rust are 
encouraged, but not required, to open a PR against [rustfmt](https://github.com/rust-lang/rustfmt) suggesting an initial
```

formatting behavior, or formatting may later be implemented as a PR by anyone,
pending approval of the implementation from T-rustfmt. T-style should be
notified to approve the interim style proposed by these PRs, but this decision
is not binding and may be revisited until the feature is stabilized and the

> formatting is codified in the style guide. 

Reviewer: I think this covers the last conversation we had at the meeting today in that the rules aren't finalized until `feature.stabilized && style_guide.contains(rules)`

After reflecting a bit more on this, I think I'm on the same page as everyone else. I could understanding wanting to get style guide PRs merged ahead of time for various reasons (e.g. ensuring it's done and done before the stable release with the stabilized feature, bors queue times, etc.), so I think that makes sense.

I just wouldn't want to see us getting into the habit of merging style guide rules for syntax that was still a long ways out from stabilization (i.e. i'd probably object to merging a style guide pr for syntax that was generally believed to be 6 months away from potential stabilization)

> Much like breaking nightly feature changes in the Rust compiler, changes should
> be not done unnecessarily, and should take into account the feature's adoption
> and readiness for stabilization. However, changes may be done until the feature

Reviewer: Another one I don't feel strongly about, just food for thought with somewhat wordier phrasing to expand upon motivation and implications around a threshold for change "necessity"

```suggestion
Much like breaking nightly feature changes in the Rust compiler, any changes
to formatting behavior for nightly syntax should be made cautiously and with
thorough consideration to avoid churn. Changes should not be done unnecessarily,
and should take into account the feature's adoption
and readiness for stabilization. However, changes may be done until the feature
```

is stabilized for various reasons: new understanding of the feature's usage in
the language, recommendation from T-style, or changes in the implementation of
the feature.

> Feature stabilization should be blocked on confirmation and codification of
> formatting bheavior. At this point, T-style may also propose alternative

Reviewer: I'm trying to recall whether t-style wanted to only block on the rules being codified in the style guide, or both codified rules _and_ rustfmt implementation being completed.

I also don't recall if we discussed whether either needs explicit buy-in from t-lang.

And one typo:

```suggestion
Feature stabilization should be blocked on confirmation and codification of
formatting behavior. At this point, T-style may also propose alternative
```

Author: > or both codified rules and rustfmt implementation being completed.

I think that should be handled between T-rustfmt and T-lang specifically.

> I also don't recall if we discussed whether either needs explicit buy-in from t-lang.

I can open an issue on the T-lang repo about T-style blocker though.

Reviewer: any thoughts/objections to me proposing this as an agenda item for this week's meet, and then creating an action item for this?

for my .02 we could go ahead and merge as-is (barring objections/requested changes from others) and then if necessary revisit and adjust the text if the t-lang conversations necessitate doing so
